Enable e2e GHA test

naveenrajm7 opened the pull request:
<!-- Thanks for sending a pull request! -->

# Changes




<!-- 
Describe your changes here- ideally you can get that description straight from
your descriptive commit message(s)! 

- :gift: Add new feature
- :bug: Fix bug
- :broom: Update or clean up current behavior
- :wastebasket: Remove feature or internal logic
-->

- Reduce knative serving components resources to fit in 2 core machine. 
- Enable GHA for e2e test

<!--
In addition, categorize the changes you're making using the "/kind" Prow command, example:

/kind <kind>

Supported kinds are: api-change, bug, cleanup, deprecation, removal, documentation, enhancement, performance

-->
/kind <kind>

<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #392

<!-- Please include the 'why' behind your changes if no issue exists -->

**Release Note**

<!--
:page_facing_up: If this change has user-visible impact, write a release note in the block
below. Include the string "action required" if additional action is required of
users switching to the new release, for example in case of a breaking change.

Write as if you are speaking to users, not other Knative contributors. If this
change has no user-visible impact, no release-note is needed.
-->
```release-note
Reduces resources of Serving components 
```

**Docs**

<!--
:book: If this change has user-visible impact, link to an issue or PR in
https://github.com/knative/docs.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [knative/docs]: <issue or pr link>
- [Feature Track]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->

psschwei commented on .github/workflows/kind-e2e.yaml:
```diff
@@ -12,8 +12,8 @@ env:
   # thus allowing us to test without bypassing tag-to-digest resolution.
-  KIND_VERSION: 0.14.0
-  KNATIVE_VERSION: 1.6.0
+  KIND_VERSION: 0.20.0
```
We should probably use the same kind versions both in this test and [as a minimum install version](https://github.com/knative-sandbox/kn-plugin-quickstart/blob/7549e95ec3beaafe44f2f32ed80c2ca64cba1767/pkg/kind/kind.go#L31), so may want to update the other place too...

psschwei commented on .github/workflows/kind-e2e.yaml:
```diff
@@ -22,47 +22,47 @@ jobs:
-    #     artifact-name: logs-${{ matrix.k8s-version}}-${{ matrix.ingress }}-${{ matrix.test-suite }}
+    #     echo "github action runner broke this test"
+
+    - name: Set up Go 1.18.x
```
This probably needs a bump to v1.20 (which I think is what we're building the rest of the project with)

psschwei commented:
/ok-to-test

naveenrajm7 commented:
Addresses comments, fixed failing e2e test. 

Currently list of deployments and amount of resources are hard-coded . if I there are other ways to explore let me know.

/re-test

skonto commented on pkg/install/install.go:
```diff
@@ -219,6 +223,21 @@ func retryingApply(path string) error {
 
+func reduceResources() error {
+	var err error
+	err = runCommand(exec.Command("kubectl", "set", "resources", "deployment", "activator", "--requests=cpu=150m,memory=30Mi", "--limits=cpu=500m,memory=300Mi", "-n", "knative-serving"))
```
Kind node seems to share the host cpu and mem so given that the sum of the Serving deployments cpu requests (including Kourier) are ~=1cpu should we try set a resourceQuota on the serving ns? Another thing to explore is limit [Kind resource allocation.](https://github.com/kubernetes-sigs/kind/issues/1578#issuecomment-631140100) if we want to use on worker node (because of the quickstart). At the Serving side we use  `kind-worker-count=4` and setup Kind with `chainguard-dev/actions/setup-kind@main`.

naveenrajm7 replied:
Thank you for this valuable suggestion. If I can limit resource usage from the kind point of view itself, then I can avoid hard reduction of deployment resources.

psschwei commented:
Test failure is legit ... we don't have a matrix in this action

naveenrajm7 commented:
> Test failure is legit ... we don't have a matrix in this action

Didn't change this part of the workflow, will check why the matrix keyword was used in the first place.

psschwei commented on pkg/install/install.go:
```diff
@@ -126,6 +126,10 @@ func Serving() error {
 		return fmt.Errorf("wait: %w", err)
 	}
 
+	if err := reduceResources(); err != nil {
+		return fmt.Errorf("reduce: %w", err)
+	}
+
```
should we also run this for eventing?  otherwise, I think this is looking pretty good
